browser/resources/settings/brave_clear_browsing_data_dialog/brave_clear_browsing_data_dialog_behavior.ts

```diff
@@ -1,151 +1,163 @@
+      this.braveRewardsEnabled_ = enabled
     })
   }
 
   override connectedCallback() {
-    super.connectedCallback();
+    super.connectedCallback()
 
-    this.onSelectedPageChangedCallback_ = this.onSelectedPageChanged_.bind(this);
+    this.onSelectedPageChangedCallback_ = this.onSelectedPageChanged_.bind(this)
     this.$.pages.addEventListener('iron-select',
-      this.onSelectedPageChangedCallback_);
+      this.onSelectedPageChangedCallback_)
 
-    this.updateSaveButtonStateCallback_ = this.updateSaveButtonState_.bind(this);
-    this.shadowRoot.querySelector('#on-exit-tab').addEventListener(
-      'clear-data-on-exit-page-change', this.updateSaveButtonStateCallback_);
+    this.updateSaveButtonStateCallback_ = this.updateSaveButtonState_.bind(this)
+    this.shadowRoot!.querySelector('#on-exit-tab')!.addEventListener(
+      'clear-data-on-exit-page-change', this.updateSaveButtonStateCallback_)
 
-    this.saveOnExitSettingsCallback_ = this.saveOnExitSettings_.bind(this);
-    this.shadowRoot.querySelector('#saveOnExitSettingsConfirm').addEventListener(
-      'click', this.saveOnExitSettingsCallback_);
+    this.saveOnExitSettingsCallback_ = this.saveOnExitSettings_.bind(this)
+    this.shadowRoot!.querySelector('#saveOnExitSettingsConfirm')!
+      .addEventListener('click', this.saveOnExitSettingsCallback_)
 
-    this.onClearBraveAdsDataClickHandler_ = this.clearBraveAdsData_.bind(this);
+    this.onClearBraveAdsDataClickHandler_ = this.clearBraveAdsData_.bind(this)
   }
 
   override disconnectedCallback() {
     super.disconnectedCallback()
 
     if (this.onSelectedPageChangedCallback_) {
       this.$.pages.removeEventListener('selected-item-changed',
-        this.onSelectedPageChangedCallback_);
-      this.onSelectedPageChangedCallback_ = null;
+        this.onSelectedPageChangedCallback_)
+      this.onSelectedPageChangedCallback_ = null
     }
 
-    if (this.saveOnExitSettingsCallback_) {
-      this.shadowRoot.querySelector('#on-exit-tab').removeEventListener(
-        'clear-data-on-exit-page-change', this.updateSaveButtonStateCallback_);
-      this.updateSaveButtonStateCallback_ = null;
+    if (this.updateSaveButtonStateCallback_) {
+      this.shadowRoot!.querySelector('#on-exit-tab')!.removeEventListener(
+        'clear-data-on-exit-page-change', this.updateSaveButtonStateCallback_)
+      this.updateSaveButtonStateCallback_ = null
     }
 
     if (this.saveOnExitSettingsCallback_) {
-      this.shadowRoot.querySelector('#saveOnExitSettingsConfirm').removeEventListener(
-        'click', this.saveOnExitSettingsCallback_);
-      this.saveOnExitSettingsCallback_ = null;
+      this.shadowRoot!.
+        querySelector('#saveOnExitSettingsConfirm')!.
+        removeEventListener('click', this.saveOnExitSettingsCallback_)
+      this.saveOnExitSettingsCallback_ = null
     }
 
-    this.onClearBraveAdsDataClickHandler_ = () => {};
+    this.onClearBraveAdsDataClickHandler_ = () => {}
```

Comment:
optional as not introduced in this PR, but if we insist on doing things this way this should probably be nulled out too - it shouldn't be possible to click anything when while the component isn't in the dom :shrug:

Reply:
See above.